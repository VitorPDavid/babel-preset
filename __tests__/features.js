import { transformAsync } from '@babel/core'
import preset from '../src'

const options = {
    presets: [preset]
}

describe('optional chaining', () => {
    it('should transform properly', async () => {
        const input = `const A = null; const B = A?.toFixed(2) || 3; B;`

        const { code: output } = await transformAsync(input, {
            ...options,
            filename: 'file.ts'
        })
        expect(output).toMatchSnapshot()

        const evaluated = eval(output)

        expect(evaluated).toEqual(3)
    })
})
