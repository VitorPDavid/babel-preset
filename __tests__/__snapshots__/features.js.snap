// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features null coalesce 1`] = `
"\\"use strict\\";

var A = null;
A !== null && A !== void 0 ? A : 2;"
`;

exports[`features optional chaining 1`] = `
"\\"use strict\\";

var A = null;
var B = (A === null || A === void 0 ? void 0 : A.toFixed(2)) || 3;
B;"
`;
